charts/graph-network-indexer/templates/indexer-service/deployment.yaml

```diff
@@ -50,49 +57,57 @@ spec:
               valueFrom:
                 fieldRef:
                   fieldPath: metadata.name
-          {{- with (deepCopy $.Values.indexerDefaults.env | mustMerge $values.env) }}
-          {{- range $key, $val := .}}
-          {{- if not (kindIs "invalid" $val) }}
+          {{- if $values.env }}
+          {{- range $key, $val := ( omit $values.env "INDEXER_SERVICE_DATABASE__POSTGRES_URL" ) }}
             - name: {{ $key | quote }}
               value: {{ $val | quote }}
           {{- end }}
           {{- end }}
-          {{- end }}
-          {{- with (deepCopy $.Values.indexerDefaults.secretEnv | mustMerge $values.secretEnv) }}
-          {{- range $key, $val := .}}
+          {{- if $values.secretEnv }}
+          {{- range $key, $val := ( omit $values.secretEnv "INDEXER_SERVICE_DATABASE__POSTGRES_URL" ) }}
             - name: {{ $key | quote }}
               valueFrom:
                 secretKeyRef:
                   name: {{ $val.secretName | quote }}
                   key: {{ $val.key | quote }}
-                  optional: false
-          {{- end }}
-          {{- end }}
-          args:
-          {{- with (deepCopy $.Values.indexerDefaults.config | mustMerge $values.config) }}
-          {{- range $key, $val := .}}
-          {{- if not (kindIs "invalid" $val) }}
-            - --{{ $key }}={{ $val }}
-          {{- end }}
           {{- end }}
           {{- end }}
+            - name: INDEXER_SERVICE_DATABASE__POSTGRES_URL
+              value: 'postgres://$(POSTGRES_USERNAME):$(POSTGRES_PASSWORD)@{{ index $values.postgresConfig "host" }}:{{ index $values.postgresConfig "port" }}/{{ index $values.postgresConfig "database" }}'          
+          command:
+          - /usr/local/bin/service
+          - --config=/etc/indexer-rs/config.toml
           {{- with $values.extraArgs }}
             {{- toYaml (. | default list) | nindent 12 }}
           {{- end }}
           ports:
             - name: http-queryapi
-              containerPort: 7600
+              containerPort: {{ index $values.service.ports "http-queryapi" }}
               protocol: TCP
+            {{- if $values.metrics.enabled }}
             - name: http-metrics
-              containerPort: 7300
+              containerPort: {{ $values.metrics.port }}
               protocol: TCP
+            {{- end }}
+          volumeMounts:
+          - name: config
+            mountPath: "/etc/indexer-rs"
+            readOnly: true
           readinessProbe:
             httpGet:
               port: http-queryapi
           {{- with $values.resources }}
           resources:
             {{- toYaml . | nindent 12 }}
           {{- end }}
+      volumes:
+        - name: config
+          configMap:
+            name: {{ include "graph-network-indexer.fullname" . }}-{{ $componentName }}-config
+        {{- if $values.initContainers }}
+        - name: init-data
```

Comment:
haven't spotted this volume being used anywhere, what is the objective of this volume?